internal/sysfs/dirfs.go

```diff
@@ -42,11 +42,7 @@ func (d *dirFS) Open(name string) (fs.File, error) {
 
 // OpenFile implements FS.OpenFile
 func (d *dirFS) OpenFile(path string, flag int, perm fs.FileMode) (fs.File, error) {
-	if f, err := platform.OpenFile(d.join(path), flag, perm); err != nil {
```

Comment:
wrapping in the syscall abstraction worked but not for all cases. it is now handled where the other OS logic is.